File: spark/src/test/scala/geotrellis/spark/io/file/FileSpatialSpec.scala
package geotrellis.spark.io.file

import geotrellis.raster.Tile
import geotrellis.spark._
import geotrellis.spark.io._
import geotrellis.spark.io.index._
import geotrellis.spark.testfiles.TestFiles


class FileSpatialSpec
    extends PersistenceSpec[SpatialKey, Tile, TileLayerMetadata[SpatialKey]]
    with SpatialKeyIndexMethods
    with TestEnvironment
    with TestFiles
    with AllOnesTestTileSpec {
  lazy val reader = FileLayerReader(outputLocalPath)
  lazy val writer = FileLayerWriter(outputLocalPath)
  lazy val deleter = FileLayerDeleter(outputLocalPath)
  lazy val copier = FileLayerCopier(outputLocalPath)
  lazy val mover  = FileLayerMover(outputLocalPath)
  lazy val reindexer = FileLayerReindexer(outputLocalPath)
  lazy val updater = FileLayerUpdater(outputLocalPath)
  lazy val tiles = FileValueReader(outputLocalPath)
  lazy val sample = AllOnesTestFile

  describe("Filesystem layer names") {
    it("should not throw with bad characters in name") {
      val layer = AllOnesTestFile
      val layerId = LayerId("Some!layer:%@~`{}id", 10)

      println(outputLocalPath)
      writer.write[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId, layer, ZCurveKeyIndexMethod)
      val backin = reader.read[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId)
    }
  }
}

File: geotrellis-python/tests/geotrellis/spark/io/file/FileSpatialSpec.py
from spec import Spec
from tests.geotrellis.spark.io.PersistenceSpec import PersistenceSpec
from tests.geotrellis.spark.io.SpatialKeyIndexMethods import _SpatialKeyIndexMethods
from tests.geotrellis.spark.TestEnvironment import _TestEnvironment
from tests.geotrellis.spark.testfiles.TestFiles import _TestFiles
from geotrellis.spark.SpatialKey import SpatialKey
from geotrellis.raster.Tile import Tile
from geotrellis.spark.TileLayerMetadata import TileLayerMetadata
from geotrellis.spark.io.file.FileLayerReader import FileLayerReader
from geotrellis.spark.io.file.FileLayerWriter import FileLayerWriter
from geotrellis.spark.LayerId import LayerId
from geotrellis.spark.io.index.ZCurveKeyIndexMethod import ZCurveKeyIndexMethod

class FileSpatialSpec(
        Spec,
        PersistenceSpec[SpatialKey, Tile, TileLayerMetadata],
        _SpatialKeyIndexMethods,
        _TestEnvironment,
        _TestFiles):

    @property
    def reader(self):
        return FileLayerReader(self.outputLocalPath)

    @property
    def writer(self):
        return FileLayerWriter(self.outputLocalPath)

    class inner(object):
        "Filesystem layer names"

        def test_bad_characters(self):
            "should not throw with bad characters in name"
            layer = self.AllOnesTestFile
            layerId = LayerId("Some!layer:%@~`{}id", 10)
            print(self.outputLocalPath)
            self.writer.write(SpatialKey, Tile, TileLayerMetadata, layerId, layer, ZCurveKeyIndexMethod)
            backin = self.reader.read(SpatialKey, Tile, TileLayerMetadata, layerId)
